Frameworks/CoreFoundation/PlugIn.subproj/CFBundle_Locale.c

```diff
@@ -663,8 +673,66 @@ static CFStringRef _CFBundleCopyLanguageFoundInLocalizations(CFArrayRef localiza
     return NULL;
 }
 
+// WINOBJC: Helper functions for workaround in _CFBundleCreateMutableArrayOfFallbackLanguages
+static CFStringRef _copyStringTruncated(CFStringRef localization, CFRange cutoff) {
+    CFMutableStringRef truncatedString = CFStringCreateMutableCopy(NULL, 0, localization);
+    CFStringDelete(truncatedString, CFRangeMake(cutoff.location, CFStringGetLength(truncatedString) - cutoff.location));
+    return truncatedString;
+}
+
+static CFStringRef _copyStringWithUnderscores(CFStringRef localization) {
+    CFMutableStringRef underscoredString = CFStringCreateMutableCopy(NULL, 0, localization);
+    CFStringFindAndReplace(underscoredString, CFSTR("-"), CFSTR("_"), CFRangeMake(0, CFStringGetLength(underscoredString)), 0);
+    return underscoredString;
+}
+
 // Given a list of localizations (e.g., provided as argument to API, or present as .lproj directories), return a mutable array of localizations in preferred order. Returns NULL if nothing is found.
 static CFMutableArrayRef _CFBundleCreateMutableArrayOfFallbackLanguages(CFArrayRef availableLocalizations, CFArrayRef preferredLocalizations) {
+// WINOBJC: The API that performs the work described below does not exist in our 3rd party libraries
+#if DEPLOYMENT_TARGET_WINDOWS
+    // Here we need to intersect the preferred languages with the available localizations
+    // We know the user languages are in preferred order, add to this list in this order
+    // Prefer the full language locale, attempt to convert any hyphens to underscores as
+    // Windows language settings are retrieved with hyphens while underscores are commonly used for localization.
+    // Finally, attempt to truncate any underscores from the language to find a base localization.
```

Comment:
nit: give examples of these. "The user's preferred language list comes in as en-US and de-DE, ..."